handlers/discount-api/src/discountEndpoint.ts

```diff
@@ -50,21 +80,49 @@ export const applyDiscountEndpoint = async (
 ) => {
 	const zuoraClient = await ZuoraClient.create(stage);
 
-	const { subscription, discount, dateToApply } = await getDiscountToApply(
-		stage,
-		headers,
-		subscriptionNumber,
-		zuoraClient,
-	);
-	return await applyDiscount(
+	const { subscription, discount, dateToApply, account } =
+		await getDiscountToApply(stage, headers, subscriptionNumber, zuoraClient);
+	console.log('Apply a discount to the subscription');
+	const discounted = await addDiscount(
 		zuoraClient,
 		subscriptionNumber,
-		subscription.termStartDate,
-		subscription.termEndDate,
-		dateToApply,
+		dayjs(subscription.termStartDate),
+		dayjs(subscription.termEndDate),
+		dayjs(dateToApply),
 		discount.productRatePlanId,
+	);
+
+	if (discounted.success) {
+		console.log('Discount applied successfully');
+	} else {
+		throw new Error('discount was not applied: ' + JSON.stringify(discounted));
+	}
+
+	const billingPreviewAfter = await getBillingPreview(
+		zuoraClient,
+		dayjs().add(13, 'months'),
```

Comment:
What's the significance of the 13 here? Would it be worth pulling out to a const to give it a name?

Reply:
not sure as that was moved code rather than something I added.  I  wonder if in turn it was moved from elsewhere.

I think traditionally we would get 13 months to be sure we cover at least 2 payments even if it's an annual subscription.

In this case 12 would be enough as we only need the next payment, not the next two.

I'll add a comment I think, as I can't think of a useful name
```suggestion
		dayjs().add(13, 'months'), // 13 months gives us minimum 2 payments even on an Annual sub
```